Goon Chemistry [FINISHED]

Iamgoofball opened the pull request:
This is a port of Goonstation's Chemicals and Recipes, originally made by Cogwerks, a super rad dude.
- Adds 14 medicines!
- Adds 3 pyrotechnics!
- Adds 3 drugs with actual gameplay uses!
- Adds 7 toxins!
- Adds a Chemical Heater for some new recipes!
- Adds Chemical Patches for touch based applications!
- Adds a Traitor Poison Kit!
- Adds Morphine Medipens!
- Adds Morphine bottles to Cargo!

Iamgoofball committed:
Goon Chemistry Commit 1

Iamgoofball committed:
Goon Chemistry Commit 2

Iamgoofball commented:
okay so apparently I can't get a number or any good reliable info on the deal with goonchem ports and goon coders and i got told by one of them that they don't have a copyright on ideas

so goonchem is back

paprka commented on code/modules/reagents/Chemistry-Goon-Medicine.dm:
```diff
@@ -0,0 +1,366 @@
+datum/reagent/morphine
+	name = "Morphine"
+	id = "morphine"
+	description = "100% chance per cycle of healing 1 point each of OXY and TOX damage."
```
You can make some morphine medipens since they're basically epipens anyway and this would be fun if this reagent jumped them out of crit temporarily too (when the reagent runs out, if they're still -100 damage, they'd fall into crit again)

Iamgoofball replied:
imo morphine replaced inaprov on goon anyways, i'll do that

Iamgoofball replied:
this has been accomplished

Iamgoofball committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into byond

Conflicts:
	code/modules/reagents/Chemistry-Reagents.dm

ike709 commented:
welcome back

Steelpoint commented:
I think that this whole business of porting another servers chemical system is just a desperate attempt to make up for our lacklustre chemistry system and trying to avoid working on a new one, though I digress.

If we do go ahead with this, we should gut our current chem system and replace all of the goon  medicine names with their closet /tg/ equivalent. This is to we can do our best to ease everyone from one chem system to another.

Iamgoofball committed:
Goon Chemistry Commit 3: Pyrotechnics and other changes

Iamgoofball committed:
Changes up the Chem Heater.

Iamgoofball committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into byond

Conflicts:
	code/modules/reagents/Chemistry-Machinery.dm
	code/modules/reagents/reagent_containers/glass/bottle.dm

Iamgoofball committed:
Fuck you merge conflicts

Iamgoofball committed:
Makes CLF3 work, fixes the Heater, adds Chemical Patches

Iamgoofball committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into byond

Iamgoofball committed:
Goon Chemistry Commit 4

Adds epipen as per request by @quiltyquilty

Quick-fix for Krokodil making you into a zombie, your skin falls off, you don't become a zombie

Adds a Morphine medipen as per request of @paprika

Accidentally added frost oil bottles back in while trying to fix a merge conflict, whoops.

forgot to remove cyanide since we replaced it with the goon variant

Fixes and Tweaks

RemieRichards commented:
Squash this.

Iamgoofball commented:
we may have a problem

the commits are up to several months old

is it possible to squash still?

RemieRichards commented:
You should still be able to squash or rebase.

Iamgoofball commented:
@RemieRichards i can't squash the "Merge branch 'master'" commits help what do I do

Iamgoofball commented:
i'm sorry but I can't squash this

i tried rebasing, squashing, and all sorts of shit

it's not squashable without a brand new pr, branch, and copypasting everything over from this PR which will mean losing all the comments

tkdrg commented:
Yeah, merge commits are usually not squashable. If you'd rebased to fix conflicts it would work but oh well. Anyway, should I remove the Do Not Merge tag?

Iamgoofball commented:
yes, this is in a finished state now

Fox-McCloud commented:
@Steelpoint 

I'd stick with realistic drug names; Star Trek chem names don't really do anything for anyone other than established chemists. Yes, I realize it'll be a rocky transition, but Goon had to go through the same exact thing--after a few months people quickly adjust.

Also morphine just lets you ignore health slowdowns while putting you to sleep (like sleep toxin) if it's been in your system too long; Saline-Glucose--which prevents certain "in crit" effects from stacking up (not present on TG--nearest equal would be reducing losebreath) and has a 33% chance to heal 1 brute and 1 burn.

I'll look over this later, but there's a few inconsistencies/inaccuracies--but it overall looks pretty good.

joonoro commented on code/game/objects/items/weapons/cigs_lighters.dm:
```diff
@@ -114,6 +114,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	..()
 	flags |= NOREACT // so it doesn't react until you light it
 	create_reagents(chem_volume) // making the cigarette a chemical holder with a maximum volume of 15
+	reagents.add_reagent("nicotine", 15)
```
Sorry, but wouldn't this mean you can no longer dunk cigarettes in other chemicals since they start off full?

Iamgoofball replied:
I'll swap it to 10 so you can still dunk them.

Jalleo replied:
@Iamgoofball would you for a later PR be willing to make a e-cig version that holds slightly more chems and gets more consumed each tick?

Miauw62 replied:
Just increase the chem capacity.
Op 12-jan.-2015 14:54 schreef "Iamgoofball" <notifications@github.com>:

> In code/game/objects/items/weapons/cigs_lighters.dm
> <https://github.com/tgstation/-tg-station/pull/6137#discussion-diff-22790615>
> :
>
> > @@ -114,6 +114,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
> >  	..()
> >  	flags |= NOREACT // so it doesn't react until you light it
> >  	create_reagents(chem_volume) // making the cigarette a chemical holder with a maximum volume of 15
> > +	reagents.add_reagent("nicotine", 15)
>
> I'll swap it to 10 so you can still dunk them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/6137/files#r22790615>.
>

Steelpoint commented:
If this is merged, at the very least @iamgoofball should change our wiki's chemistry page to goonstation's one in the interim.

Fox-McCloud commented:
@Miauw62 

Alternatively, there could be "DIY Smokes Kit" that has no reagents in it--for example, on Goon, there's regular cigs (nicotine), nicotine-free cigs (capsaicin), and pro-puffs (contain random reagents...ranging from toxic to healing)..

Lumipharon commented:
you mean like, rollies, available from cigarette vendors?

Fox-McCloud commented:
@Lumipharon 

Yes, but only no ambrosia involved.

Iamgoofball committed:
Increases the cigarette chemical cap as per request by @Miauw62

Iamgoofball committed:
Requested changes by @Cheridan

Iamgoofball committed:
Adds a changelog and reduces drug text spam.

Iamgoofball commented:
lowered the spam amount of drugs
changes rng bottle to a set poison kit with all the traitor toxins in it
removes copypasta
adds a changelog

Iamgoofball committed:
Whoops, gotta adjust the changelog again.

tkdrg commented on code/modules/mob/living/living.dm:
```diff
@@ -275,6 +275,10 @@ Sorry Giacom. Please don't be mad :(
+/mob/living/proc/getTotalLoss(var/amount)
+	amount = getBruteLoss() + getOxyLoss() + getToxLoss() + getFireLoss() + getCloneLoss() + getBrainLoss()
+	return amount
+
```
Why are you adding this proc, it's unused as far as I can see. Also it makes no sense to include getBrainLoss here.

Fox-McCloud commented:
I'll second what tkdrg said---even in the context of Goon's system, this doesn't make any sense.

If you're going for the brain damage = lethal thing , then do what Goon does, and have it check for brain damage--if it hits 100, then have the mob become paralyzed/weakened and accumulate losebreath until they die or their brain damage drops below 100.

Regardless, changes may be needed to be made to the cloner to prevent mid-clone death.

Iamgoofball commented:
@Fox-McCloud i removed getTotalLoss(), i don't quite realize why I had it there

tkdrg commented on code/modules/reagents/Chemistry-Goon-Medicine.dm:
```diff
@@ -0,0 +1,335 @@
+		M.emote("scream")
+	if(method == INGEST)
+		M.adjustToxLoss(0.5*volume)
+		M << "You probably shouldn't have eaten that. Maybe you should of splashed it on, or applied a patch?"
```
Grammar: `Maybe you should have splashed it, or applied a patch?`

Iamgoofball committed:
span classes

Iamgoofball committed:
forgot to end a span class

tkdrg commented on code/modules/reagents/Chemistry-Goon-Medicine.dm:
```diff
@@ -0,0 +1,335 @@
+	if(!M) M = holder.my_atom
+	if(method == TOUCH)
+		M.heal_organ_damage(volume*REM,0)
+		M.adjustFireLoss((volume-(volume*2)*REM))
```
Is this equation really correct? It looks really weird.

Fox-McCloud replied:
I'd like to add that synthflehs is meant to be heal 1.5 brute and fire both per unit applied and not 1; it also stops bleeding (though I realize that isn't a thing on TG and is reserved purely to Bay and Goon).

Iamgoofball replied:
oh, it's 1.5? huh, the wiki said 1.

also yeah the equation works somehow

tkdrg commented on code/modules/reagents/Chemistry-Goon-Medicine.dm:
```diff
@@ -0,0 +1,335 @@
+datum/reagent/diphenhydramine
+	name = "Diphenhydramine"
+	id = "diphenhydramine"
+	description = "Stun reduction per cycle, increases run speed slightly, minor stamina regeneration buff, stabilizes crit."
```
This description is wrong.

Iamgoofball replied:
descriptions do not show up in game in any mean so I got lazy with updating them. imo they should be scrapped entirely but that's for a separate PR

tkdrg commented on code/modules/reagents/Chemistry-Goon-Medicine.dm:
```diff
@@ -0,0 +1,335 @@
+		if(I)
+			M.drop_item()
+	cycle_count++
+	..()
```
This is different from the description above. One of the two is wrong.

Iamgoofball replied:
descriptions do not show up in game in any mean so I got lazy with updating them. imo they should be scrapped entirely but that's for a separate PR

Fox-McCloud commented on code/modules/reagents/Chemistry-Goon-Medicine.dm:
```diff
@@ -0,0 +1,335 @@
+		M << "<span class='notice'>You probably shouldn't have eaten that. Maybe you should of splashed it on, or applied a patch?</span>"
+	..()
+	return
+
```
Silver sulfa and Styptic both heal 1 damage per volume and not 2.

The tox damage is a once off thing, which you seem to have gotten down.

The per cycle effects of styptic and silver sulfa both have a 55% chance of healing 2 of their respective damage type; this is particularly useful for hyposprays on Goon as they avoid INGEST and it allows these chems to be used as unreliable injectables.

Iamgoofball replied:
how that equation works:

let's say we have a volume of 50

it's 50-(50*2)

since adjusting by a negative value heals, it will heal 50

Fox-McCloud commented on code/modules/reagents/Chemistry-Goon-Medicine.dm:
```diff
@@ -0,0 +1,335 @@
+		M.radiation = 0
+	..()
+	return
+
```
Chacoal heals 1.5 tox not 2.0

Also don't forget about the purgative effects of charcoal and pentetic acid; charcoal increases the metabolization_rate of ALL chems (minus itself) by 0.5. Pentetic increases the metabolization_rate of ALL chems (minus itself) by 4.

Iamgoofball replied:
chemical purging was broken and has been scrapped for the current build, it will be readded if I can get a working version of it that doesn't require intense snowflake.

Iamgoofball replied:
actually I'm working on chem purging now

Iamgoofball replied:
chem purging added

Fox-McCloud commented on code/modules/reagents/Chemistry-Goon-Medicine.dm:
```diff
@@ -0,0 +1,335 @@
+	M.jitteriness -= 1
+	..()
+	return
+
```
Diphenhydramine's biggest selling point is that that is purges histamine; it increases the metabolization_rate of histamine by 3.

Also, it causes drowsyness, not reduces it =p

Iamgoofball replied:
shite, wrong flag there

also yeah I'll make it purge histamine

tkdrg commented on code/modules/reagents/Chemistry-Machinery.dm:
```diff
@@ -1238,3 +1258,63 @@ obj/machinery/computer/pandemic/proc/replicator_cooldown(var/waittime)
+	var/energy = 50
+	var/max_energy = 50
+	var/amount = 30
+	var/obj/item/weapon/reagent_containers/beaker = null
```
Don't you need to `new` this beaker?

tkdrg commented on code/modules/reagents/reagent_containers/glass/bottle.dm:
```diff
@@ -48,6 +48,16 @@
+
+	New()
+		..()
+		reagents.add_reagent("morphine", 30)
```
Please make this one use the new `spawned_reagent` system as well.

tkdrg commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -100,11 +101,10 @@
 	..()
-	reagents.remove_reagent("inaprovaline", 10)
-	reagents.add_reagent("leporazine", 9)
 	reagents.add_reagent("lexorin", 1)
```
You're going to add 10 leporazine and 1 lexorin into a container that only fits 10 units. This won't work like this.

tkdrg commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -113,11 +113,22 @@
 	..()
-	reagents.remove_reagent("inaprovaline", 10)
-	reagents.add_reagent("hyperzine", 9)
 	reagents.add_reagent("lexorin", 1)
```
Same as above.

Iamgoofball committed:
Adds an extra chemical, tweaks, fixes, changes to be more accurate to Goonchem.

Aurx commented on code/modules/reagents/Chemistry-Goon-Medicine.dm:
```diff
@@ -0,0 +1,346 @@
+datum/reagent/perfluorodecalin
+	name = "Perfluorodecalin"
+	id = "perfluorodecalin"
+	description = "Heals 25 OXY damage, but you can't talk. 33% chance of healing 1 BRUTE and 1 BURN."
```
The chemmaster can show you the desc of any chem you put in it.

Iamgoofball committed:
Updates the changelog.

Iamgoofball committed:
fixes medipens more

Iamgoofball committed:
fixes medipens even more

Iamgoofball committed:
removes scrollbar from chemdispenser nanoui menu due to new chems

Iamgoofball committed:
removes getTotalLoss() for REAL THIS TIME

tkdrg commented:
Some concerns (for the future):
1) Make the chem heater have an actual HTML interface with user feedback on the heating progress.
2) Make the chem heater researchable/constructable and upgradable.

Jalleo commented:
@Jalleo issue report what tkdrg said.

Fox-McCloud commented:
Yeah, currently the chem heater is ...really wonky; the temperature ranges from 1 to 3000 (Goon chem only needs 0 to 1000); likewise, with Goonchem you can slap the beaker in and set the temperature, then start heating/cooling and watch the temperature slowly go up/down...this is especially important in a few chemical reactions like Crank---where improperly making it will blow up in your face.

pudl commented:
wiki updated!
